src/main/resources/errors.yaml

```diff
@@ -45,31 +45,125 @@
+    body: |-
+      All primary key columns must be provided when inserting a document into a table. 
+      
+      The table ${keyspace}.${table} defines the primary key columns: ${primaryKeys}.
+      
+      The request included values for primary key columns: ${providedKeys}.
+      The request did not include values for primary key columns: ${missingKeys}.
+      
+      Resend the request with the missing primary key columns.
+
+  - scope: DOCUMENT
+    code: UNKNOWN_TABLE_COLUMNS
+    title: Unknown table columns in document
+    body: |-
+      Only columns defined in the table schema can be included when inserted a document into a table.
+      
+      The table ${keyspace}.${table} defines the columns: ${allColumns}.
+      The request included the following unknown columns: ${unknownColumns}.
+      
+      Resend the request using only defined columns.
+
+  - scope: DOCUMENT
+    code: UNSUPPORTED_COLUMN_TYPES
+    title: Unsupported table column data types in document
+    body: |-
+      Only supported column types can be included when inserting a document into a table.
+      
+      The table ${keyspace}.${table} defines the columns: ${allColumns}.
+      The request included the following columns that have unsupported data types: ${unsupportedColumns}.
+      
+      Resend the request using only supported column types.
+
+  # FILTER request errors
+  - scope: FILTER
+    code: INVALID_FILTER
+    title: Invalid filter
+    body: |-
+      The filter clause is invalid. ${filter}
+      
       ${SNIPPET.CONTACT_SUPPORT}
+ # FILTER request errors
+  - scope: FILTER
+    code: MULTIPLE_ID_FILTER
+    title: the title for the error
+    body: |-
+      big long message with ${vars} in it
+
+# ================================================================================================================
+# Server Errors
+# ================================================================================================================
 
 server-errors:
+  # UNSCOPED server errors
+  - scope:
+    code: UNEXPECTED_SERVER_ERROR
+    http-status-override: 500
+    title: Unexpected server error
+    body: |-
+      An unexpected server error occurred while processing the request. 
+      
+      Error Class: ${errorClass}
+      Error Message: ${errorMessage}
+
+  # DATABASE scope server errors
   - scope: DATABASE
-    code: SERVER_READ_FAILED
-    title: the title for the error
-    body: big long body with ${vars} in it
+    code: CLOSED_CONNECTION
+    title: Database connection was closed while processing the request
+    body: |-
+      The Data API connection to the database was closed by the database while processing the request.
+      
+      Writing to the ${schemaType} ${keyspace}.${table} failed to complete successfully. If this request modified data the changes may have been written to by some replicas, but not all. Future read requests may return eventually consistent results.
```

Comment:
We'll probably want to "snipettify" explanation of eventual consistency.

Reply:
def will do when we get into these further.